Tools.cs

```diff
@@ -245,5 +245,12 @@ public void ResetPlayerPosition()
         
         Debug.Log($"Teleporting to {pos.x}, {pos.y}, {pos.z}");
         world.PlaceCurrentPlayerAt(pos, rot, true);
-    } 
+    }
+
+    public void RefilPlayerBoost()
+    {
+        var player = world.GetCurrentPlayer();
```

Comment:
Could you add `if (!coreHasBeenSetup) return;` above this line? We set `coreHasBeenSetup` after we find the right references within the scene, so if we access `world` while it's not setup, it'll throw a NullReferenceException. This mostly happens between levels.

Otherwise everything looks great!

Reply:
Sure ! Should be fixed by commit [e83361a](https://github.com/Lagaffe91/SpeedUtils/commit/e83361ad37f666b0687bcd956036395167cc0b78). 👍